Gordon360/Controllers/ProfilesController.cs

```diff
@@ -476,6 +485,37 @@ public async Task<ActionResult<FacultyStaffProfileViewModel>> UpdateOfficeHours(
             return Ok(result);
         }
 
+        /// <summary>
+        /// Set visibility of some piece of personal data for user.
+        /// </summary>
+        /// <param name="userPrivacy">Faculty Staff Privacy Decisions (see UserPrivacyViewModel)</param>
+        /// <returns></returns>
+        [HttpPut]
+        [Route("user_privacy")]
+        // [StateYourBusiness(operation = Operation.UPDATE, resource = Resource.PROFILE_PRIVACY)]
+        public async Task<ActionResult<UserPrivacyViewModel>> UpdateUserPrivacyAsync(UserPrivacyViewModel userPrivacy,string authenticatedUserUsername)
+        {
+            //var authenticatedUserUsername = AuthUtils.GetUsername(User);
+            await _profileService.UpdateUserPrivacyAsync(authenticatedUserUsername, userPrivacy);
+
+            return Ok();
+        }
+
+        /// <summary>
+        /// Return a list buildings.
+        /// </summary>
+        /// <returns> All buildings</returns>
+        [HttpGet]
+        [Route("visibility_group")]
+        public ActionResult<IEnumerable<string>> GetVisibilityGroup()
+        {
+            var groups = _context.UserPrivacy_Visibility_Groups.Select(fs => fs.Group)
+                                   .Distinct()
+                                   .Where(d => d != null)
+                                   .OrderBy(d => d);
```

Comment:
```suggestion
                                   .Where(d => d != null)
                                   .OrderBy(d => d);
```
not sure what `d` is supposed to be. My first assumption is that `d` stands for `distinct` but that doesn't tell me much either.

Reply:
I copied this code from another file😂. You're right. I will change the `d` to `g` which stands for `groups` and makes more sense.